fix(insights): table selector options sometimes doesn't reset with actions selector

DominikB2014 opened the pull request:
Full details in DAIN-244

It's possible that when you change database systems in the db module (screenshot 1) , the list of tables in the table selector (screenshot 2) might combine between the two database systems. In otherwords, you can end up with mongodb tables in the table selector, even if the postgresql system is selected.

To resolve this, the compact select cache component now takes in a cache key, this key will used to cache the results of the mongodb table results separately from the postgresql results. The cache key will just be the `additionalQuery`, which is the `span.system:db` part of the full query. The module only let's you select one system or the other, but not both, therefore we never have to cache them together.

<img width="530" height="207" alt="image" src="https://github.com/user-attachments/assets/fc6d5297-4c09-4811-b615-d9165acd3530" />
<img width="559" height="389" alt="image" src="https://github.com/user-attachments/assets/4764f9cf-ed3e-423b-8424-eb92b0d0c5a4" />

DominikB2014 committed:
add cache key to compact select options cache

DominikB2014 committed:
fix tests

DominikB2014 committed:
update

gggritso reviewed (commented):
This makes sense, but I'm wary of adding another way to clear the option cache. `useCompactSelectOptionsCache` returns a `clearCache` function, and it's already in use in the domain selector. Is it possible to use that instead? Right now it already [clears the cache if the project IDs change](https://github.com/getsentry/sentry/blob/f309777a1577f4111205d7cc3f94cf32145a94a0/static/app/views/insights/common/views/spans/selectors/domainSelector.tsx#L145-L147)

DominikB2014 commented:
> This makes sense, but I'm wary of adding another way to clear the option cache. `useCompactSelectOptionsCache` returns a `clearCache` function, and it's already in use in the domain selector. Is it possible to use that instead? Right now it already [clears the cache if the project IDs change](https://github.com/getsentry/sentry/blob/f309777a1577f4111205d7cc3f94cf32145a94a0/static/app/views/insights/common/views/spans/selectors/domainSelector.tsx#L145-L147)

Yeah we could do that, that just means we have to refetch things everytime the cache clears (I think the initial queries will be cached, but anything that was appended onto the list will need to be refetched)

DominikB2014 commented:
> @DominikB2014 I guess so, but regardless, it's not a good idea to have competing mechanisms for cache management. I like the cache key approach, but in that case would you mind removing the current cache clearing logic and replacing it with cache keys?

@gggritso agreed and will do!

DominikB2014 committed:
remove unused import

DominikB2014 committed:
fix

DominikB2014 commented:
@gggritso I removed the cache clearing logic and replaced with cache keys. 

One thing to note, i didn't remove the ability to clear the cache per key in the hook itself, i feel as though that isn't competing functionality, just standard functionality for any caching hook.

gggritso reviewed (approved):
Great, thanks!

DominikB2014 committed:
Merge branch 'master' into dominikbuszowiecki/dain-244-queries-mismatched-table-options-in-dropdown
